rs-react-app/src/api/pokeapi.ts

```diff
@@ -0,0 +1,41 @@
+export interface PokemonListItem {
+  name: string;
+  url: string;
+}
+
+export interface PokemonDetails {
+  name: string;
+  sprites: {
+    front_default: string;
+  };
+  types: { type: { name: string } }[];
+}
+
+export async function fetchPokemonList(
+  offset = 0,
+  limit = 20
+): Promise<PokemonListItem[]> {
+  const response = await fetch(
+    `https://pokeapi.co/api/v2/pokemon?offset=${offset}&limit=${limit}`
```

Comment:
The API URL appears multiple times in the code. It makes sense to use a variable instead of a magic string.